### PR TITLE
feat: Introduce C# class library exposing `Config` related implementation.

### PR DESCRIPTION
**Pull request by paulhazen**
This PR introduces a C# class library based on the source files within the plugin. It does not export anywhere or do anything right now, but eventually we will leverage this class library in concert with another project within the solution to generate a native code library that deals with the config stuff using the implementation from the plugin itself.

#EOS-2231

### CODE REVIEW

File: lib/NativeCode/DynamicLibraryLoaderHelper/EOSPluginConfig/com.playeveryware.eos/Runtime/Core/Application.cs [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+namespace PlayEveryWare.EpicOnlineServices
+{
+    internal class Application
+    {
+        public const string streamingAssetsPath = "C:/Users/PaulPEW/dev/repos/eos_plugin_for_unity/Assets/StreamingAssets/";
```
Comment (WispyMouse) [inline lib/NativeCode/DynamicLibraryLoaderHelper/EOSPluginConfig/com.playeveryware.eos/Runtime/Core/Application.cs]: This is a suspicious line.

**Review by WispyMouse — CHANGES_REQUESTED** [review]
Looks mostly good, though it does have a constant pointing to your own computer drive.

I want to run this in local before signing off.

File: lib/NativeCode/DynamicLibraryLoaderHelper/EOSPluginConfig/com.playeveryware.eos/Runtime/Core/Application.cs (hunk 1 above)
Comment (paulhazen) [reply lib/NativeCode/DynamicLibraryLoaderHelper/EOSPluginConfig/com.playeveryware.eos/Runtime/Core/Application.cs]: Yep! Good call. I'll fix this in the AM

File: com.playeveryware.eos/Runtime/EOS_SDK/Core/Helper.cs [hunk 2]
```diff
@@ -630,6 +630,11 @@ private static IntPtr AddPinnedBuffer(Utf8String str)
 			return AddPinnedBuffer(str.Bytes, 0);
 		}
 
+        internal static IntPtr AddPinnedBuffer(byte[] array)
+        {
+            return AddPinnedBuffer(new ArraySegment<byte>(array));
+        }
```
Comment (paulhazen) [inline com.playeveryware.eos/Runtime/EOS_SDK/Core/Helper.cs]: This should have the `// PEW Edit start` and end comments.

Comment (paulhazen) [reply com.playeveryware.eos/Runtime/EOS_SDK/Core/Helper.cs]: Also, we should include the `null` test inside the function.

File: lib/NativeCode/DynamicLibraryLoaderHelper/DynamicLibraryLoaderHelper.sln [hunk 3]
```diff
@@ -1,12 +1,14 @@
 ﻿
 Microsoft Visual Studio Solution File, Format Version 12.00
-# Visual Studio Version 16
-VisualStudioVersion = 16.0.31025.194
+# Visual Studio Version 17
+VisualStudioVersion = 17.8.34408.163
```
Comment (paulhazen) [inline lib/NativeCode/DynamicLibraryLoaderHelper/DynamicLibraryLoaderHelper.sln]: These should stay the same for the time being.

File: lib/NativeCode/DynamicLibraryLoaderHelper/DynamicLibraryLoaderHelper.sln [hunk 4]
```diff
@@ -1,12 +1,14 @@
 ﻿
 Microsoft Visual Studio Solution File, Format Version 12.00
-# Visual Studio Version 16
-VisualStudioVersion = 16.0.31025.194
+# Visual Studio Version 17
```
Comment (arthur740212) [inline lib/NativeCode/DynamicLibraryLoaderHelper/DynamicLibraryLoaderHelper.sln]: version probably should not be changed yet

**Review by arthur740212 — APPROVED** [review]

**Comment by WispyMouse** [comment]
FileSystemUtility.cs had this error while trying to build:

> Cannot define a class or member that utilizes tuples because the compiler required type 'System.Runtime.CompilerServices.TupleElementNamesAttribute' cannot be found. Are you missing a reference?	EOSPluginConfig	C:\SNGH\eos_plugin_for_unity_clone\com.playeveryware.eos\Runtime\Core\Utility\FileSystemUtility.cs	488

Can you address that?

**Review by WispyMouse — APPROVED** [review]
Before merging, please address the FileSystemUtility Tuple problem.

**Comment by paulhazen** [comment]
> Before merging, please address the FileSystemUtility Tuple problem.

Done. Could not replicate the issue after I reverted some of the version stuff.

**Pull request merged**